features/mbedtls/mbed_lib.json

```diff
@@ -0,0 +1,157 @@
+            "value": 0
+        },
+        "ECDH_ECDSA_WITH_AES_256_GCM_SHA384": {
+            "help": "Include support for the TLS-ECDH-ECDSA-WITH-AES-256-GCM-SHA384 ciphersuite",
+            "value": 0
+        },
+        "PSK_WITH_AES_128_GCM_SHA256": {
+            "help": "Include support for the TLS-PSK-WITH-AES-128-GCM-SHA256 ciphersuite",
+            "value": 0
+        },
+        "PSK_WITH_AES_256_GCM_SHA384": {
+            "help": "Include support for the TLS-PSK-WITH-AES-256-GCM-SHA384 ciphersuite",
+            "value": 0
+        },
+        "PSK_WITH_AES_256_CBC_SHA384": {
+            "help": "Include support for the TLS-PSK-WITH-AES-256_CBC-SHA384 ciphersuite",
+            "value": 0
+        },
+        "ECDHE_RSA_WITH_AES_128_CBC_SHA256": {
+            "help": "Include support for the TLS-ECDHE-RSA-WITH-AES-128-CBC-SHA256 ciphersuite",
+            "value": 0
+        },
+        "ECDHE_RSA_WITH_AES_128_GCM_SHA256": {
+            "help": "Include support for TLS-ECDHE-RSA-WITH-AES-128-GCM-SHA256 ciphersuite",
+            "value": 0
+        },
+        "ECDHE_RSA_WITH_AES_256_CBC_SHA384": {
+            "help": "Include support for the TLS-ECDHE-RSA-WITH-AES-256-CBC-SHA384 ciphersuite",
+            "value": 1
+        },
+        "ECDHE_RSA_WITH_AES_256_GCM_SHA384": {
+            "help": "Include support for the TLS-ECDHE-RSA-WITH-AES-256-GCM-SHA384 ciphersuite",
+            "value": 0
+        },
+        "RSA_WITH_AES_256_GCM_SHA384": {
+            "help": "Include support for the TLS-RSA-WITH-AES-256-GCM-SHA384 ciphersuite",
+            "value": 0
+        },
+        "RSA_WITH_AES_128_GCM_SHA256": {
+            "help": "Include support for the TLS-RSA-WITH-AES-128-GCM-SHA256 ciphersuite",
+            "value": 0
+        },
+        "RSA_WITH_AES_128_CBC_SHA256": {
+            "help": "Include support for the TLS-RSA-WITH-AES-128-CBC-SHA256 cipersuite",
+            "value": 0
+        },
+        "RSA_WITH_AES_256_CBC_SHA256": {
+            "help": "Include support for the TLS-RSA-WITH-AES-256-CBC-SHA256 cipersuite",
+            "value": 0
+        },
+        "ECDH_RSA_WITH_AES_128_CBC_SHA256": {
+            "help": "Include support for TLS-ECDHE-RSA-WITH-AES-128-GCM-SHA256 ciphersuite",
+            "value": 0
+        },
+        "ECDH_RSA_WITH_AES_128_GCM_SHA256": {
+            "help": "Include support for the TLS-ECDH-RSA-WITH-AES-128-GCM-SHA256 ciphersuite",
+            "value": 0
+        },
+        "ECDH_RSA_WITH_AES_256_CBC_SHA384": {
+            "help": "Include support for the TLS-ECDH-RSA-WITH-AES-256-CBC-SHA384 ciphersuite",
+            "value": 0
+        },
+        "ECDH_RSA_WITH_AES_256_GCM_SHA384": {
+            "help": "Include support for the TLS-ECDH-RSA-WITH-AES-256-GCM-SHA384 ciphersuite",
+            "value": 0
+        }
+    },
+    "macros": [ "MBEDTLS_USER_CONFIG_FILE=\"mbedtls_config_reader.h\"",
```

Comment:
These macros at the moment can't be overwritten by an app (limitation of the config). Is that intention here?

cc @theotherjimmy

Reply:
That's the implementation here. I can't speak to intentions.

Reply:
That's the intention. They are generally needed regardless of the config the user wants.